### PR TITLE
Docker cleanup

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Our experience from other projects (e.g. the new fun-mooc), leads us to consider cleaning our docker environments and definitions. Our goal: one single Dockerfile and one single docker-compose definition, because: `ETOOMANYFILES`.

## Proposal

- [x] make the development image a build stage instead of a separated Dockerfile
- [x] remove unused alpine images
- [x] remove docker-compose test environment
- [x] refactor db engine switch

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -79,14 +77,61 @@ COPY docker/files/usr/local/etc/gunicorn/richie.py /usr/local/etc/gunicorn/richi
...
+    --output dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz \
+    https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && \
+    tar -C /usr/local/bin -xzvf dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz && \
+    rm dockerize-linux-amd64-$DOCKERIZE_VERSION.tar.gz
+
+# Restore the un-privileged user running the application
+ARG DOCKER_USER
+USER ${DOCKER_USER}
+
+# Target database engine & port
```
Comment (jmaupetit) [inline Dockerfile]: `s/engine/host/`

File: docker/files/usr/local/bin/entrypoint [hunk 2]
```diff
@@ -13,23 +13,23 @@
 #
 # To pass environment variables, you can either use the -e option of the docker run command:
 #
-#     docker run --rm -e USER_NAME=foo -e HOME='/home/foo' richie:latest python sandbox/manage.py migrate
+#     docker run --rm -e USER_NAME=foo -e HOME='/home/foo' richie:latest python manage.py migrate
```
Comment (sampaccoud) [inline docker/files/usr/local/bin/entrypoint]: :medal_sports:

File: docker-compose.yml [hunk 3]
```diff
@@ -1,54 +1,48 @@
...
     ports:
         - "5440:5432"
 
+  mysql:
+    image: mysql:5.6
+    env_file:
+      - env.d/development/mysql
+    ports:
+      - "3316:3306"
+    command: mysqld --character-set-server=utf8 --collation-server=utf8_general_ci
```
Comment (sampaccoud) [inline docker-compose.yml]: IMO this is a regression... running 2 db containers on your system seems like a burden.

Comment (jmaupetit) [reply docker-compose.yml]: Only one will! Look at the app service dependencies!

Comment (lunika) [reply docker-compose.yml]: all the magic also happen in the makefile. By running only the `app` container.

Comment (jmaupetit) [reply docker-compose.yml]: ...and it's dynamic dependencies!

**Pull request merged**